### PR TITLE
Add types for react-switch-case

### PR DESCRIPTION
**Pull request by Falci**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
... removed

If removing a declaration:
... removed

### CODE REVIEW

**Comment by Falci** [comment]
What's necessary to merge this PR?

File: types/react-switch-case/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export interface SwitchProps {
+    condition: any;
+}
+
+interface CaseProps {
+    value: any;
+}
+
+declare class Swtich extends React.Component<SwitchProps> {}
```
Comment (sandersn) [inline types/react-switch-case/index.d.ts:16]:
```suggestion
declare class Switch extends React.Component<SwitchProps> {}
```

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good except for the typo

**Comment by sandersn** [comment]
@Falci we switch the bot to a webhook instead of a reguarly-scheduled task and only now fixed the code that brings new packages to the DT maintainer's attention. Sorry for the wait.

**Review by sandersn — APPROVED** [review]

**Comment by Falci** [comment]
Ready to merge

**Pull request merged**